routers/api/v1/repo/repo.go

```diff
@@ -500,6 +500,289 @@ func GetByID(ctx *context.APIContext) {
 	ctx.JSON(200, repo.APIFormat(perm.AccessMode))
 }
 
+// Edit edit repository properties
+func Edit(ctx *context.APIContext, opts api.EditRepoOption) {
+	// swagger:operation PATCH /repos/{owner}/{repo} repository repoEdit
+	// ---
+	// summary: Edit a repository's properties. Only fields that are set will be changed.
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo to edit
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo to edit
+	//   type: string
+	//   required: true
+	//   required: true
+	// - name: body
+	//   in: body
+	//   description: "Properties of a repo that you can edit"
+	//   schema:
+	//     "$ref": "#/definitions/EditRepoOption"
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/Repository"
```

Comment:
You should add some more responses e.g. 403 forbidden

Reply:
A lot of the API endpoints don't do that, but will do.

Reply:
Ah, I see that those that only have InternalServerError only have the 201 or 200.

Reply:
Ok, added.

Reply:
Yeah, if you don't put the 403 etc. when you generate a swagger API you won't get them differentiated nicely.

Reply:
Have added these.